negotiation/neural/models.py

```diff
@@ -304,13 +304,21 @@ def forward(self, tgt, memory_bank, state, memory_lengths=None):
         # Check
         assert isinstance(state, RNNDecoderState)
         tgt_len, tgt_batch = tgt.size()
-        _, memory_batch, _ = memory_bank.size()
+        if isinstance(memory_banks, list):
+            memory_bank = memory_banks[0]
+        else:
+            # We transpose since the code originally did so when calculating
+            # attention in StdRNNDecoder._run_forward_pass(), but we can no
+            # longer do that anymore because memory_banks is sometimes a list
+            memory_banks = memory_banks.transpose(0, 1)
+            memory_bank = memory_banks
```

Comment:
this part is a bit sketchy. they use time major (seq_len as first dimension) for all inputs. let's not change that.